### ISSUE TITLE
cant compile desmume but compile the fork "desmume reloaded" fine

### ISSUE THREAD

**Issue by CaioVdmm** [TASVideos/desmume#305]
I cant compile Desmume due to an glib erro on main.cpp an it IS a desmume falt but @zeromus  refuse to help and always close mais posts without investigate the error .
my two last posts is #304 and #302 this two posts is about this error but @zeromus close all of than claiming that is my system fault but "desmume reloaded" than have the same code of desmume version of 2018 compile fine so it IS DESMUME Fault its not my system falt 
so the error is in #304 and #302

**Comment by zeromus** [comment]
duplicate of #302
Stop making new threads or I will ban you.
You don't know anything about software engineering and you arent competent to judge which software's fault this is. 
Your error logs are useless (you have deleted too much of them due to not understanding what's important)
Your conclusion about the relevance of 2018 desmume reloaded to desmume 2019 master is irrelevant. Software can perfectly correctly do things that smoke out errors in archlinux. The bug isn't with "Software" because it smoked it out but with the true bug, which is most likely a corrupted but on your hard disk putting garbage before _Pragma

**Comment by CaioVdmm** [comment]
and how i cam compile Desmume reloaded fine if he have the same code??

**Comment by zeromus** [comment]
desmume reloaded doesnt smoke out the bad bit on your disk.

**Comment by shibotto** [comment]
>which is most likely a corrupted bit on your hard disk putting garbage before _Pragma

That is very unlikely. I get the same on my Gentoo machine however, full relevant log:

```
Making all in .
make[1]: Entering directory '/tmp/desmume/desmume/src/frontend/posix'
make[1]: Nothing to be done for 'all-am'.
make[1]: Leaving directory '/tmp/desmume/desmume/src/frontend/posix'
Making all in gtk
make[1]: Entering directory '/tmp/desmume/desmume/src/frontend/posix/gtk'
Making all in doc
make[2]: Entering directory '/tmp/desmume/desmume/src/frontend/posix/gtk/doc'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/tmp/desmume/desmume/src/frontend/posix/gtk/doc'
make[2]: Entering directory '/tmp/desmume/desmume/src/frontend/posix/gtk'
depbase=`echo main.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
g++ -DPACKAGE_NAME=\"desmume\" -DPACKAGE_TARNAME=\"desmume\" -DPACKAGE_VERSION=\"svn\" -DPACKAGE_STRING=\"desmume\ svn\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"desmume\" -DVERSION=\"svn\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DSVN_REV=0 -DSVN_REV_STR=\"0\" -DHAVE_LIBZ=1 -DHAVE_GL_GL_H=1 -DHAVE_GL_GLU_H=1 -DHAVE_LIBDL=1 -DHAVE_LIBGL=1 -DHAVE_GL_GLX_H=1 -DHAVE_GL_GLX=1 -DHAVE_LIBSOUNDTOUCH=1 -DHAVE_JIT=1 -DHOST_LINUX=1 -DHOST_64=1 -I.  -I../../../../src/ -I../../../../src/libretro-common/include -I../../../../src/frontend -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/lib64/libffi-3.3_rc0/include -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/uuid -I/usr/include/libdrm -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/atk-1.0 -pthread  -pthread -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include  -I/usr/include/alsa   -I/usr/include/soundtouch    -g -O2 -MT main.o -MD -MP -MF $depbase.Tpo -c -o main.o main.cpp &&\
mv -f $depbase.Tpo $depbase.Po
In file included from /usr/lib64/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:32,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from main.cpp:28:
/usr/include/glib-2.0/glib/gmacros.h:988:33: error: missing binary operator before token "_Pragma"
  988 | #define _GLIB_GNUC_DO_PRAGMA(x) _Pragma(G_STRINGIFY (x))
      |                                 ^~~~~~~
/usr/include/glib-2.0/glib/gmacros.h:989:31: note: in expansion of macro '_GLIB_GNUC_DO_PRAGMA'
  989 | #define GLIB_DEPRECATED_MACRO _GLIB_GNUC_DO_PRAGMA(GCC warning "Deprecated pre-processor symbol")
      |                               ^~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/gversionmacros.h:418:49: note: in expansion of macro 'GLIB_DEPRECATED_MACRO'
  418 | # define GLIB_DEPRECATED_MACRO_IN_2_32          GLIB_DEPRECATED_MACRO
      |                                                 ^~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/deprecated/gthread.h:270:38: note: in expansion of macro 'GLIB_DEPRECATED_MACRO_IN_2_32'
  270 | #define g_thread_supported()     (1) GLIB_DEPRECATED_MACRO_IN_2_32
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
main.cpp:3610:6: note: in expansion of macro 'g_thread_supported'
 3610 | #if !g_thread_supported()
      |      ^~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtkobject.h:37,
                 from /usr/include/gtk-2.0/gtk/gtkwidget.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkcontainer.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkbin.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkwindow.h:36,
                 from /usr/include/gtk-2.0/gtk/gtkdialog.h:35,
                 from /usr/include/gtk-2.0/gtk/gtkaboutdialog.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:33,
                 from main.cpp:28:
/usr/include/gtk-2.0/gtk/gtktypeutils.h:236:64: warning: 'GTypeDebugFlags' is deprecated [-Wdeprecated-declarations]
  236 | void            gtk_type_init   (GTypeDebugFlags    debug_flags);
      |                                                                ^
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from main.cpp:28:
/usr/include/glib-2.0/gobject/gtype.h:685:3: note: declared here
  685 | } GTypeDebugFlags GLIB_DEPRECATED_TYPE_IN_2_36;
      |   ^~~~~~~~~~~~~~~
In file included from /usr/include/gtk-2.0/gtk/gtktoolitem.h:31,
                 from /usr/include/gtk-2.0/gtk/gtktoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtkmenutoolbutton.h:30,
                 from /usr/include/gtk-2.0/gtk/gtk.h:126,
                 from main.cpp:28:
/usr/include/gtk-2.0/gtk/gtktooltips.h:73:12: warning: 'GTimeVal' is deprecated: Use 'GDateTime' instead [-Wdeprecated-declarations]
   73 |   GTimeVal last_popdown;
      |            ^~~~~~~~~~~~
In file included from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glib-2.0/gobject/gbinding.h:28,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:30,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from main.cpp:28:
/usr/include/glib-2.0/glib/gtypes.h:549:26: note: declared here
  549 | typedef struct _GTimeVal GTimeVal GLIB_DEPRECATED_TYPE_IN_2_62_FOR(GDateTime);
      |                          ^~~~~~~~
make[2]: *** [Makefile:597: main.o] Error 1
make[2]: Leaving directory '/tmp/desmume/desmume/src/frontend/posix/gtk'
make[1]: *** [Makefile:661: all-recursive] Error 1
make[1]: Leaving directory '/tmp/desmume/desmume/src/frontend/posix/gtk'
make: *** [Makefile:1801: all-recursive] Error 1
```

Using Glib 2.62.3 and GCC 9.2.0
